pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/BuildConditionResponderTest.java

```diff
@@ -242,4 +242,21 @@ public void changedAndNotSuccess() throws Exception {
         j.assertLogNotContains("I FAILED", b3);
     }
 
+    @Issue("JENKINS-44993")
+    @Test
+    public void failureInPostBlock() throws Exception {
+        expect(Result.FAILURE, "failureInPostBlock")
+                .logContains("[Pipeline] { (foo)",
+                        "hello",
+                        "[Pipeline] { (" + SyntheticStageNames.postBuild() + ")",
+                        "I FAILED",
+                        "Error when executing failure post condition",
+                        "Error when executing always post condition",
+                        "No such property: otherUndefined for class",
+                        "No such property: undefined for class")
```

Comment:
I'd expect this line to occur immediately after "Error when executing always post condition",

Reply:
See above.